pkg/tecdsa/dkg/member.go

```diff
@@ -256,9 +259,382 @@ func (fm *finalizingMember) MarkInactiveMembers(
+		// Check if message from self.
+		if message.senderID == sm.index {
+			continue
+		}
+
+		// Check if sender sent multiple messages.
+		if duplicatedMessagesFromSender(message.senderID) {
+			sm.logger.Infof(
+				"[member: %v] received multiple messages from sender: [%d]",
+				sm.index,
+				message.senderID,
+			)
+			continue
+		}
+
+		// Sender's preferred DKG result hash doesn't match current member's
+		// preferred DKG result hash.
+		if message.resultHash != sm.preferredDKGResultHash {
+			sm.logger.Infof(
+				"[member: %v] signature from sender [%d] supports result different than preferred",
+				sm.index,
+				message.senderID,
+			)
+			continue
+		}
+
+		// Check if the signature is valid.
+		ok, err := signing.VerifyWithPublicKey(
+			message.resultHash[:],
+			message.signature,
+			message.publicKey,
+		)
+		if err != nil {
+			sm.logger.Infof(
+				"[member: %v] verification of signature from sender [%d] failed: [%v]",
+				sm.index,
+				message.senderID,
+				err,
+			)
+			continue
+		}
+		if !ok {
+			sm.logger.Infof(
+				"[member: %v] sender [%d] provided invalid signature",
+				sm.index,
+				message.senderID,
+			)
+			continue
+		}
+
+		receivedValidResultSignatures[message.senderID] = message.signature
+	}
+
+	// Register member's self signature.
+	receivedValidResultSignatures[sm.index] = sm.selfDKGResultSignature
+
+	return receivedValidResultSignatures, nil
+}
+
+// shouldAcceptMessage indicates whether the given member should accept
+// a message from the given sender.
+func (sm *signingMember) shouldAcceptMessage(
+	senderID group.MemberIndex,
+	senderPublicKey []byte,
+) bool {
+	isMessageFromSelf := senderID == sm.index
+	isSenderValid := sm.membershipValidator.IsValidMembership(
+		senderID,
+		senderPublicKey,
+	)
+	isSenderAccepted := sm.group.IsOperating(senderID)
+
+	return !isMessageFromSelf && isSenderValid && isSenderAccepted
+}
+
+// SubmissionConfig contains parameters describing DKG submission process.
+type SubmissionConfig struct {
+	// GroupSize is the size of a group in TBTC.
+	GroupSize int
+	// HonestThreshold is the minimum number of active participants behaving
+	// according to the protocol needed to generate a signature.
+	HonestThreshold int
+	// ResultPublicationBlockStep is the duration (in blocks) that has to pass
+	// before group member with the given index is eligible to submit the
+	// result.
+	// Nth player becomes eligible to submit the result after
+	// T_dkg + (N-1) * T_step
+	// where T_dkg is time for phases 1-12 to complete and T_step is the result
+	// publication block step.
+	ResultPublicationBlockStep uint64
+}
+
+// SubmittingMember represents a member submitting a DKG result to the
+// blockchain along with signatures received from other group members supporting
+// the result.
+type submittingMember struct {
+	logger log.StandardLogger
+
+	// Represents the member's position for submission.
+	index  group.MemberIndex
+	config *SubmissionConfig
```

Comment:
Can't we just embed the `*signingMember` here and expose `initializeSubmittingMember` from there to do the conversion just like we do in other DKG states?

Reply:
Done in be88ead76ae30ea4f9dd01d5b2f00342e374b986.